stubs/pytest-lazyfixture/pytest-lazy-fixture.pyi

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Iterable, List
+
+class LazyFixture:
+    name: str
+    def __init__(self, name: str) -> None: ...
+    def __eq__(self, other: LazyFixture) -> bool: ...
+
+def lazy_fixture(names: str | Iterable[str]) -> LazyFixture | List[LazyFixture]: ...
```

Comment:
Union return types are annoying to work with. For example, you would need to use `assert isinstance(foo, LazyFixture)` or similar to convince mypy that it can't be a list. Use an overload instead:

```suggestion
@overload
def lazy_fixture(names: str) -> LazyFixture: ...
@overload
def lazy_fixture(names: Iterable[str]) -> list[LazyFixture]: ...
```

You will need to add `from typing import overload`. Also, `list` instead of `List` is not a typo; it's newer syntax, and we now prefer it.

Reply:
thanks a lot